physlr/physlr.py

```diff
@@ -1332,6 +1332,79 @@ def determine_molecules_bc_cosine_of_squared(g, u):
                 for community in
                 Physlr.detect_communities_cosine_of_squared(g, bi_connected_component)]
 
+    @staticmethod
+    def partition_subgraph_into_bins_randomly(node_set, max_size=40):
+        """Prtition the subgraph into bins randomly for faster processing. Return bins."""
+        bins_count = 1
+        if len(node_set) > max_size:
+            bins_count = 1 + int(len(node_set) / max_size)
+        node_list = list(node_set)
+        random.shuffle(node_list)
+        size, leftover = divmod(len(node_set), bins_count)
+        bins = [node_list[0 + size * i: size * (i + 1)] for i in list(range(bins_count))]
+        edge = size * bins_count
+        for i in list(range(leftover)):
+            bins[i % bins_count].append(node_list[edge + i])
+        bin_sets = [set() for _ in range(len(bins))]
+        for i, c in zip(range(len(bins)), bins):
+            bin_sets[i].update(set(c))
```

Comment:
```suggestion
            bin_sets[i].update(c)
```
Is `set` needed here? Avoid unnecessary type conversions.

Reply:
Yes. The standard for all the functions that I am writing for `determine_molecules` is list of sets. Rather than consistency, It makes it so easy to build new pipelines for molecule separation based on these functions.

Reply:
`update` takes an enumeration as input, so the type that contains the enumeration doesn't affect the result. Both `bin_sets[i].update(set(c))` and `bin_sets[i].update(c)` yield the same result.

Reply:
Oh I see.. I did set `bin_sets = [set() for _ in bins]` so it is already a list of sets and... got it. Thanks